packages/nx/src/plugins/js/project-graph/build-dependencies/build-dependencies.ts

```diff
@@ -12,8 +25,23 @@ export function buildExplicitDependencies(
 ): RawProjectGraphDependency[] {
   if (totalNumberOfFilesToProcess(ctx) === 0) return [];
 
+  const externalDependenciesCache: ExternalDependenciesCache = new Map();
   let dependencies: RawProjectGraphDependency[] = [];
 
+  /**
+   * NOTE: It is important that we process package.json files first so that accurate versions for
```

Comment:
Doing this first... probably shouldn't cause any issues.. but I think it's weird that it is important for this to be first...

The 2 parts should be independent :thinking: 

Especially because someone can have `analyzePackageJson` false? :thinking: